core-common/src/main/java/org/zalando/nakadi/repository/kafka/KafkaLocationManager.java

```diff
@@ -68,7 +89,12 @@ private void updateBootstrapServers(final boolean createWatcher) {
             }
             for (final String brokerId : childrenBuilder.forPath(BROKERS_IDS_PATH)) {
                 final byte[] brokerData = curator.getData().forPath(BROKERS_IDS_PATH + "/" + brokerId);
-                brokers.add(Broker.fromByteJson(brokerData));
+                if (this.kafkaSettings.getPreferredListenerPort().isPresent()) {
```

Comment:
to avoid duplication and make it even better to read we could do:
`brokers.add(Broker.fromByteJson(brokerData, kafkaSettings.getPreferredListenerPort()));`

and  `Broker.fromByteJson` looks like 
```
        static Broker fromByteJson(final byte[] data, final Optional<Integer> portOpt)
....
            return new Broker(host, portOpt.orElse(json.getInt("port")));
        }
```